plugin/core/registry.py

```diff
@@ -105,8 +104,19 @@ def sessions(self, capability: Optional[str] = None) -> Generator[Session, None,
         yield from sessions_for_view(self.view, capability)
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
+class LspRestartServerCommand(LspTextCommand):
     def run(self, edit: Any) -> None:
         window = self.view.window()
+        if not window:
+            return
+
+        sessions = [session.config.name for session in self.sessions()]
+        if sessions:
+            window.show_quick_panel(sessions, lambda index: self.restart_server(index, sessions))
```

Comment:
You could pass the window through the lambda to simplify the code (no need to lookup or null-check) but no big deal.

Reply:
That means something along the lines of this? correct?

```python
window.show_quick_panel(config_names, lambda index: self.restart_server(window, index, config_names) if window else False)
```